src/omotes_orchestrator/main.py

```diff
@@ -580,14 +586,45 @@ def task_progress_update(self, progress_update: TaskProgressUpdate) -> None:
 
             if progress_update.progress == 0:  # first progress indicating calculation start
                 logger.debug("Progress update was the first. Setting job %s to RUNNING", job.id)
-                self.omotes_sdk_if.send_job_status_update(
-                    job=job,
-                    status_update=JobStatusUpdate(
-                        uuid=str(job.id),
-                        status=JobStatusUpdate.JobStatus.RUNNING,
-                    ),
-                )
-                self.postgresql_if.set_job_running(job.id)
+                amount_of_starts = self.postgresql_if.count_job_starts(job.id)
+                logger.debug("Job %s has started %s times previously.", job.id, amount_of_starts)
+                if amount_of_starts >= self.config.delivery_limit_threshold_per_job:
+                    logger.error(
+                        "Job %s has been started too many times (limit %s). Cancelling.",
+                        job.id,
+                        self.config.delivery_limit_threshold_per_job,
+                    )
+                    self.celery_if.cancel_workflow(job_db.celery_id)
+                    self.omotes_sdk_if.send_job_status_update(
+                        job=job,
+                        status_update=JobStatusUpdate(
+                            uuid=str(job.id), status=JobStatusUpdate.JobStatus.CANCELLED
```

Comment:
I am doubting whether an ERROR status would make more sense here.  
To me CANCELLED means 'cancelled by user', error means that something went wrong.

Reply:
Yeah, I was also doubting initially. My reasoning:
- `JobStatus.ERROR` means something went wrong but job is completed.
- `JobStatus.CANCELLED` means the job was not complete so no result possible. <-- Job did not complete so seemed to fit better.

Perhaps we should add a `JobStatus.UNPROCESSABLE` or something in a later version?

Reply:
I would say that ERROR means that the job is not completed but prematurely terminated, due to runtime error or for instance out-of-memory error.  

Sounds good to add another status in a later version!

Reply:
Great, made an issue for it here: https://github.com/Project-OMOTES/orchestrator/issues/67